### PR TITLE
Feature autocomplete handler

### PR DESCRIPTION
**Pull request by mayoung00**

### CODE REVIEW

**Review by gfry99 — CHANGES_REQUESTED** [review]
Couple of things I found during my first pass.

1. Make sure to make the documentation uniform, I'm seeing some issues involving capitalization at the beginning of the documentation.
2. There is currently an issue with the way we are printing  the prompt to the user, currently a user is able to hold backspace and delete some characters of the prompts.  Can we do some looking into this issue to find a potential issue.

File: UMLEditor/UMLEditor/Classes/AutoCompleteHandler.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        return LastMatchingIndex(toTest, candidate) != -1;
+    }
+
+    /// <summary>
+    /// Gets suggestions for rest of command given partial typed command
+    /// </summary>
+    /// <param name="forLine">Partially completed command from user</param>
+    /// <param name="index">Index of where user stopped typing</param>
+    /// <returns>array of possible commands</returns>
+    public string[] GetSuggestions(string forLine, int index)
```
Comment (Robunnin) [inline UMLEditor/UMLEditor/Classes/AutoCompleteHandler.cs:84]: Why is there an overload of this method where index and forLine are flipped?

File: UMLEditor/UMLEditor/Classes/AutoCompleteHandler.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                ++result;
+            }
+            else
+            {
+                break;
+            }
+        }
+
+        return result;
+    }
```
Comment (Robunnin) [inline UMLEditor/UMLEditor/Classes/AutoCompleteHandler.cs:65]: This seems to be using my initial demo implementation, which was a rough proof of concept. A cleaner way to do canidate finding would be to use the index provided in LastMatchinIndex to do size comparisions and String.StartsWith to see if the command is a match. 

For instance, if the user entered 10 characters and a command is 5 you immediately know it isn't a match. If the number of characters is <= the length of the command then if the command starts with the entered text it is a match.
Commands: add_class add_method
User entered add (4 characters)
Both start with add_: **add**_class **add**_method

**Comment by Robunnin** [comment]
The prompt bugs seems to only happen after a ctrl+backspace, which is supported in Windows shells but doesn't seem to be on bash (I have not tested macOS yet)

File: UMLEditor/UMLEditor/Views/CommandLine/CommandLineController.cs [hunk 3]
```diff
@@ -38,7 +39,6 @@ public CommandLineController()
     /// <summary>
     /// Executes command
     /// </summary>
-    /// <param name="input">command to execute</param>
```
Comment (Robunnin) [inline UMLEditor/UMLEditor/Views/CommandLine/CommandLineController.cs:41]: The tag for input got deleted in the documentation block

**Review by gfry99 — APPROVED** [review]
Merging, but keep Robert's comment [here](https://github.com/mucsci-students/2022sp-420-NotJava/pull/218/files#r838983817) in mind for a possible refactor.  Currently if a user types in list_ and cycles through possible commands, load_diagram shows up as a potential option.

**Pull request merged**